### PR TITLE
Use capact cli in all scripts

### LINKED ISSUE #328: Implement running local cluster via CLI
## Description

1. **CLI:** Implement command to quickly set up local Capact cluster. It might be split into two commands (see the "Possible options" section
1. **Website:** Use [https://github.com/asciinema/asciinema-player](https://github.com/asciinema/asciinema-player)  or [https://github.com/ines/termynal](https://github.com/ines/termynal) to display command for bootstrapping local Capact

Tasks:

- [x] Provision kind cluster - https://github.com/capactio/capact/pull/372
- [x] Install Capact
- [ ]  Use `capat env create kind` and `capact install` in our scripts

### Possible option

- Migrate our bash scripts to Go. Check before implementation if that is possible,  if not then fallback to previous idea (clone repo and use Bash scripts)
   * Clone subdirectory hack [https://stackoverflow.com/questions/600079/how-do-i-clone-a-subdirectory-only-of-a-git-repository](https://stackoverflow.com/questions/600079/how-do-i-clone-a-subdirectory-only-of-a-git-repository|smart-link)
 * Use charts from latest/stable helm repo
 * Provision command uses kind Go API
 * Git, kubectl, helm installed beforehand if not installed
    - Use SKIP_DEPS_INSTALLATION=true
    - For Git, we can check whether we can use go-git or GitHub CLI Golang API
    - Implement installation of kubectl, helm in our installation scripts
 * Add flag to specify release version (tag on git repo - by default latest release?)
 * Modify installation script to allow not building images
 * Update doc
     - development guide
     - local installation doc
 * Use it for local development
 * Use CLI on CI
    - PR: build cli from sources and  use it for integration tests
    - Branch: use CLI for integration tests

### Ideas

```bash
capact provision kind
capact install # by default latest semver git tag,
# alternatively
capact install --version @main/0.3.0/0.5.0
```

## Reason

Currently, there is no way to simply install Capact on local machine. You need to clone the repo, install required dependencies and run `make dev-cluster`. The steps should be improved and simplified.

This will help with getting users to get started quickly with the platform

 1. For development
 1. And experimentation / POC

### PR DESCRIPTION
**Pull request by lukaszo**
## Description

Changes proposed in this pull request:

- Replace bash functions and use `capact install` command in our scripts
- Allow to override any Capact, ingress-controller and cert-manager helm values

## Testing
Create cluster, test it, delete it

```shell
make dev-cluster
make dev-cluster-update
make dev-cluster-delete
```

## Related issue(s)

Closes #328

### CODE REVIEW

**Review by mszostok — COMMENTED** [review]
Tested, it works like a charm 👌 

BTW
I know that it was already merge like that, but maybe we should add version details under dedicated step? Currently, it looks strange:
Screenshot 2021-07-19 at 16 40 18

File: hack/test-integration.sh [hunk 1]
```diff
@@ -53,39 +53,27 @@ capact::test::execute() {
...
-      capact::update::images_on_kind
     fi
 
     export INCREASE_RESOURCE_LIMITS="false" # To comply with the default GitHub Actions Runner limits
-    export CLUSTER_TYPE="KIND"
+    export CLUSTER_TYPE="kind"
     export USE_TEST_SETUP="true"
     export PRINT_INSECURE_NOTES="false"
-    capact::install_upgrade::charts
+    capact::install
```
[the remark this answers is not in the document]
Comment (lukaszo) [reply hack/test-integration.sh:76]: We already have an agreement about CLI output. I will just add `helm -n capact-system get notes capact` to `dev-cluster-create.sh`  and add a link to the docs in NOTES.txt

File: hack/lib/utilities.sh [hunk 2]
```diff
@@ -516,21 +267,14 @@ capact::version_supported(){
 
 capact::validate::tools() {
```
[the remark this answers is not in the document]
Comment (lukaszo) [reply hack/lib/utilities.sh:268]: Added to `dev-cluster-create.sh` as it's using Helm now to show the notes.

**Comment by mszostok** [comment]
We also should remove redundant files from: https://github.com/capactio/capact/tree/main/hack/cluster-config/kind

to do not maintain it in two places. Also README.md needs to be adjusted. The `openssl` commands can be copied to go file where certs are embeded.

**Comment by mszostok** [comment]
Testing scenario:
```
# install capact with scripts from main

# update with Capact CLI
DOCKER_REPOSITORY=ghcr.io/capactio DOCKER_TAG=3b5b9ae BUILD_IMAGES=false DISABLE_MONITORING_INSTALLATION=true ENABLE_POPULATOR=false make dev-cluster-update
```


1. It was already merged, but we changed the release name from `ingress-nginx` to `ingress-controller`
```bash
ingress-controller-ingress-nginx-controller-8ff8b4c9f-7hh4w   0/1     Pending            0          7s
ingress-nginx-controller-758d79bf79-x9cns                     1/1     Running            0          148m
```
This installs a new release instead of updating the old one. I wasn't aware of that, do we want to keep this change?
1. ~`monitoring` was still on the components list~ - nvm, I didn't pull last changes from PR
1. The wrong repository source is displayed:
```
 ✓ Creating namespace capact-system
	Version: @local
	Helm repository: https://storage.googleapis.com/capactio-stable-charts
```
but the local one is used, it should print the `capact.LocalChartsPath` value
1. Do we want to fix that? IMO it will be worth to
Screenshot 2021-07-21 at 18 46 18

File: internal/cli/capact/components.go [hunk 3]
```diff
@@ -376,23 +399,23 @@ func (k *Kubed) InstallUpgrade(version string) (*release.Release, error) {
...
 		values, err := ValuesFromString(capactKindOverridesYaml)
 		if err != nil {
 			return nil, errors.Wrap(err, "while converting override values")
 		}
-		mappedValues = tools.MergeMaps(values, mappedValues)
+		capactValues = tools.MergeMaps(values, capactValues)
 	}
-	return c.runUpgrade(upgradeCli, mappedValues)
+
+	for _, value := range c.opts.Parameters.Override.CapactStringOverrides {
```
Comment (mszostok) [inline internal/cli/capact/components.go:412]: it's done already in `SetCapactValuesFromOverrides` do we need to repeat that?

so maybe we should get rid of `SetCapactValuesFromOverrides` or unify our approach and have same func for `cert-manager` and `ingress`

**Comment by mszostok** [comment]
As none of detected problems are critical, we can merge that and create follow-up PR with fixes if you prefer :)

File: internal/cli/capact/components.go (hunk 3 above)
Comment (lukaszo) [reply internal/cli/capact/components.go:412]: In SetCapactValuesFromOverrides we only set values in the struct CapactValues. Other values are ignored there. Some of these values are used before installing Capact.


Here we pass all values.

File: internal/cli/capact/components.go [hunk 4]
```diff
@@ -476,24 +515,25 @@ func (h *Helm) InstallComponents(w io.Writer, status *printer.Status) error {
...
-	resp, err := http.Get(CRDUrl)
-	if err != nil {
-		return errors.Wrapf(err, "while getting CRD %s", CRDUrl)
+	var reader io.Reader
+	if h.opts.Parameters.Version == LocalVersionTag {
+		f, err := os.Open(CRDLocalPath)
+		if err != nil {
+			return errors.Wrapf(err, "while opening local CRD file%s", CRDLocalPath)
+		}
+		reader = f
```
Comment (mszostok) [inline internal/cli/capact/components.go:526]:
```suggestion
		defer f.Close()
		reader = f
```

**Review by mszostok — APPROVED** [review]
🚀 

| tombstone  | tenor-79580408 |
|:---:|:---:|

**Pull request merged**